specifyweb/frontend/js_src/lib/components/Toolbar/DataModel.tsx

```diff
@@ -297,11 +311,35 @@ function DataModelRelationships({
   readonly model: SpecifyModel;
 }): JSX.Element {
   const data = React.useMemo(() => getRelationships(model), [model]);
+
+  const [filteredDependentData, setFilteredDependentData] =
+    React.useState(data);
+  const [dependentFilter, setDependentFilter] = React.useState(false);
+
+  React.useEffect(() => {
```

Comment:
Refactor to replace useEffect with useMemo - try to avoid useEffect whenever possible
See https://beta.reactjs.org/learn/you-might-not-need-an-effect for more information